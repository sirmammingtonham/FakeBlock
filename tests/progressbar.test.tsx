import React from 'react';
import LinearWithValueLabel from '../src/components/progressbar';
// import Popup from '../src/components/popup';
// import * as renderer from 'react-test-renderer';
import {shallow, configure} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';

configure({adapter: new EnzymeAdapter()});

test('Test linear value label creation when percentage as 50', () => {
	// doesnt work bruh
	// const component = renderer.create(
	// 	<LinearWithValueLabel percentage={50}/>
	// );
	// const progressbar = component.toJSON();
	// expect(progressbar).toMatchSnapshot();
});

describe('Check if the component is defined after creation', () => {
	it('should be defined', () => {
		expect(LinearWithValueLabel).toBeDefined();
	});
});

test('ProgressBar showing correct progress', () => {
	const progressbar = shallow(<LinearWithValueLabel percentage={50}/>);
	expect(progressbar.getElement().props.children.props.value).toBe(50);
});

test('ProgressBar showing correct progress', () => {
	const progressbar = shallow(<LinearWithValueLabel percentage={90}/>);
	expect(progressbar.getElement().props.children.props.value).toBe(90);
});
