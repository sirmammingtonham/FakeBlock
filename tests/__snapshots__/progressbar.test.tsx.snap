// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test linear value label creation when percentage as 50 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiBox-root MuiBox-root-2"
  >
    <div
      className="MuiBox-root MuiBox-root-3"
    >
      <div
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={50}
        className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate"
        role="progressbar"
      >
        <div
          className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
          style={
            Object {
              "transform": "translateX(-50%)",
            }
          }
        />
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-4"
    >
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        Number of fake info detected: 50
      </p>
    </div>
  </div>
</div>
`;
